src/Pomelo.EntityFrameworkCore.MySql/Storage/Internal/MySqlRelationalTransaction.cs

```diff
@@ -87,7 +88,7 @@ public void Dispose()
             if (!_disposed)
             {
                 _disposed = true;
-                if (_transactionOwned)
+                if (_transactionOwned && _dbTransaction.Connection.State == ConnectionState.Closed)
```

Comment:
I did look to catch the exception earlier in the call stack, but was unable to figure out a more elegant method.  Ideally, the inner exception should rollback any transactions and close the connection.